src/control/controlobjectscript.cpp

```diff
@@ -1,9 +1,11 @@
+#include "control/controlobjectscript.h"
+
 #include <QtDebug>
 
-#include "control/controlobjectscript.h"
+#include "controllers/controllerdebug.h"
 
 ControlObjectScript::ControlObjectScript(const ConfigKey& key, QObject* pParent)
-        : ControlProxy(key, pParent, ControlFlag::NoAssertIfMissing) {
+        : ControlProxy(key, pParent, ControllerDebug::shouldAssertForInvalidControlObjects()) {
```

Comment:
This does not assert in tests 8ec7ed8

Reply:
There is no crash. I don't know what you're talking about.

Reply:
If you're questioning the changes merged in #3056, this is not an appropriate place to do so. This PR is not changing behavior; only refactoring. I will not discuss any proposals to change behavior here. This PR has been waiting long enough. There is lots more work ahead.

Reply:
I think this will always assert not. Mapping Typo = crash ...

This is a change in this PR and that must not happen. Instead a error dialog is appropriated like we did for other scripting issues.

See related discussion.
https://mixxx.zulipchat.com/#narrow/stream/247620-development-help/topic/Crash.20starting.202.2E3.20if.20controller.20script.20has.20a.20specific.20kind

Reply:
This is not a behavior change. It was just implemented a little differently compared to main when resolving merge conflicts. [Here](https://github.com/mixxxdj/mixxx/blob/main/src/controllers/engine/controllerengine.cpp#L551) is the debug assertion in main.

Reply:
This only asserts if mixxx was started with the `--controllerDebug` flag and debug assertions are compiled in. And it will only crash if debug assertions are additionally set to fatal.

Also, this behavior was not introduced by this branch, so it shouldn't be a blocker. And any behavioral changes don't belong in a refactoring PR if it can be avoided.

Reply:
That sound reasonable. 
Unfortunately I had hard times to understand that from the code. 
I think this can be solved by a comment or a better function name. 

From a distance I think it is that the root cause is that  "shouldAssertForInvalidControlObjects =  ControlFlag::None" read "Like Don't Assert". So I get hooked to the inverted logic. 

Instead the function should become a bool. 
And than we can do here more visible. 
shouldAssertForInvalidControlObjects() ? ControlFlag::None : ControlFlag::AllowMissingOrInvalid

In addition ControlFlag::None can become ControlFlag::AssertValid or something